src/monfaction.h

```diff
@@ -2,38 +2,93 @@
 #ifndef CATA_SRC_MONFACTION_H
 #define CATA_SRC_MONFACTION_H
 
-#include <unordered_map>
+#include <cstdint>
+#include <limits>
+#include <map>
+#include <set>
+#include <vector>
 
+#include "enum_traits.h"
+#include "optional.h"
 #include "type_id.h"
 
 class JsonObject;
 
+template <typename T> class generic_factory;
+
 enum mf_attitude {
     MFA_BY_MOOD = 0,    // Hostile if angry
     MFA_NEUTRAL,        // Neutral even when angry
     MFA_FRIENDLY,       // Friendly
-    MFA_HATE            // Attacks on sight
+    MFA_HATE,           // Attacks on sight
+
+    MFA_SIZE            // reserved, last element
+};
+
+template<>
+struct enum_traits<mf_attitude> {
+    static constexpr mf_attitude last = MFA_SIZE;
 };
 
-using mfaction_att_map = std::unordered_map< mfaction_id, mf_attitude >;
+using mfaction_att_map = std::map<mfaction_str_id, mf_attitude>;
+using mfaction_att_vec = std::vector<u_int8_t>;
```

Comment:
Please do not use u_int8_t. This data type is not supported in the Windows environment. Use instead uint8_t.

Reply:
Fixed in #45507